common/types_test.go

```diff
@@ -51,3 +54,34 @@ func TestHashJsonValidation(t *testing.T) {
 		}
 	}
 }
+
+func TestAddressUnmarshalJSON(t *testing.T) {
+	var a Address
+	var tests = []struct {
+		Input     string
+		ShouldErr bool
+		Output    *big.Int
+	}{
+		{"", true, nil},
+		{`""`, true, nil},
+		{`"0x"`, true, nil},
+		{`"0x00"`, true, nil},
+		{`"0xG000000000000000000000000000000000000000"`, true, nil},
+		{`"0x0000000000000000000000000000000000000000"`, false, big.NewInt(0)},
+		{`"0x0000000000000000000000000000000000000010"`, false, big.NewInt(16)},
+	}
+	for i, test := range tests {
+		err := a.UnmarshalJSON([]byte(test.Input))
+		if err != nil && !test.ShouldErr {
+			t.Error(i, "expected no error got ", err)
```

Comment:
Please use: `t.Errorf("test #%d:  unexpected error: %v", i, err)`. This format conforms to the tests in the Go standard libraries. Also please update the other to `t.Error`-s too to use `t.Errorf`.